feat: applicant validation improvments

MaxErikssonSK opened the pull request:
## Types of changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [ ] Bug fix
- [x] New feature
- [ ] Removed feature
- [ ] Code style update (formatting etc.)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Content/Data

## Does this PR introduce a breaking change?
- [ ] Yes (I have stepped the version number accordingly)
- [x] No
     
## Checklist:

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [x] My code follows the code style of this project.
- [ ] I have updated the documentation accordingly (if applicable).
- [ ] I have added/updated tests to cover my changes (if applicable).

Jira: https://jira.sundsvall.se/browse/DRAKEN-2460

MaxErikssonSK committed:
feat: applicant validation improvments

hensankom commented on frontend/cypress/e2e/register-errand.cy.ts:
```diff
@@ -406,6 +406,13 @@ describe('Registrera ärende-sida', () => {
     cy.get('.sk-dialog').find('button').contains('Ja').click();
   });
 
+  it('prevents registration when applicant is missing', () => {
```
Hm testet/testerna går inte igenom för mig. Får detta:

<img width="788" height="86" alt="Image" src="https://github.com/user-attachments/assets/7f8bfd0f-7a59-46dd-9b59-0e2626dcf920" />

MaxErikssonSK committed:
fix: mosly test related fixes

- Cypress test updates.
- Removed redundant interface value regarding extraparameter.
- Fixed bug regarding readonly state after creating draft errands.
- Added test case for extraparameters validations to be able to register new errands.
